### PR TITLE
String iterations and methods

### PR DESCRIPTION
**Pull request by KalyaniPayyavula**
- Added string iterations
- Added unit test
- Files changed - leanString.js, learnStrings.test.js

image

### CODE REVIEW

File: learnStrings/learnString.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            mostRepChar = [char]
+        } else if (testObj[char] = mostRepCharCount) {
+            mostRepChar.push(char)
+        }
+    }
+    return mostRepChar;
+}
+// check weather a string is palindrome or not
+
+export const checkPalindrome = str => {
```
Comment (yugandhar-pathi) [inline learnStrings/learnString.js]: rename function name to `isPalindrome`.  And return boolean values.

File: learnStrings/learnString.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    let revStr = str.split("").reverse().join("")
```
Comment (yugandhar-pathi) [inline learnStrings/learnString.js]: Use `const` for variables which are not changed.

File: learnStrings/learnString.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { findMostRepChar, checkPalindrome, removeLastPath } from './learnString'
+
+test("find most repeated char in the array", () => {
+    expect(findMostRepChar("abb")).toEqual(["b"])
+    expect(findMostRepChar("abbcc")).toEqual(["b", "c"])
```
Comment (yugandhar-pathi) [inline learnStrings/learnString.test.js]: Use proper match to check for repeated char.

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

**Comment by yugandhar-pathi** [comment]
Unit test coverage is very low. update the screenshot.

File: learnStrings/learnString.js (hunk 1 above)
Comment (KalyaniPayyavula) [reply learnStrings/learnString.js]: changed

File: learnStrings/learnString.js (hunk 2 above)
Comment (KalyaniPayyavula) [reply learnStrings/learnString.js]: changed

File: learnStrings/learnString.test.js (hunk 3 above)
Comment (KalyaniPayyavula) [reply learnStrings/learnString.test.js]: done

File: learnStrings/learnString.js (hunk 1 above)
Comment (yugandhar-pathi) [reply learnStrings/learnString.js]: But function is not returning boolean values.

File: learnStrings/learnString.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        return "Given string is palindrome"
+    } else {
+        return "Given string is not palindrome"
+    }
+}
+
+// remove last path from a given url
+
+export const removeLastPath = str => {
+    let res = str.slice(0, str.lastIndexOf("/"))
```
Comment (yugandhar-pathi) [inline learnStrings/learnString.js]: You can convert this to plain arrow function right. You don't have any other logic other than return.

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: sortingArray/sortArray.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+export const sortArrays = arr => {
```
Comment (yugandhar-pathi) [inline sortingArray/sortArray.js]: Modify this function to take two args. One is `array` and the other to mention it as `asc` or `dsc`.
so this method should be something like below ...

sortArrays([1,5,2], "asc" ); //  [1,2,5]
sortArrays([1,5,2], "dsc" ); //  [5,2,1]

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: sortingArray/sortArray.js (hunk 5 above)
Comment (KalyaniPayyavula) [reply sortingArray/sortArray.js]: fixed

File: learnClass/learnClass.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+let resh = new Student("resh", 6)
+let skan = new Student("Skandhana", 1)
+let jahn = new Grade5Students("jahn", 12, 80)
+console.log(jahn.getGrade())
+console.log(resh.getGrade())
+console.log(skan.getGrade())
+
+class Animal {
```
Comment (yugandhar-pathi) [inline learnClass/learnClass.js]: Move Class : Animal, Dog & Cat to different file.

File: learnClass/learnClass.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        return 5;
+    }
+}
+
+let resh = new Student("resh", 6)
+let skan = new Student("Skandhana", 1)
+let jahn = new Grade5Students("jahn", 12, 80)
+console.log(jahn.getGrade())
+console.log(resh.getGrade())
+console.log(skan.getGrade())
```
Comment (yugandhar-pathi) [inline learnClass/learnClass.js]: Remove this code from here. Write unit tests for this.

File: learnClass/learnClass.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    makeSound() {
+        super.openMouth("Dog")
+        console.log(this.dogSound)
+    }
+}
+//let animal = new Animal("Makes sound")
+let cat = new Cat("Meow meow")
+cat.makeSound()
+let dog = new Dog("Bow bow")
+dog.makeSound()
```
Comment (yugandhar-pathi) [inline learnClass/learnClass.js]: Remove this code from here and add Unit tests.

File: learnClass/learnClass.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+// console.log(cat.doSound())
+// console.log(cat.makeSound())
+// console.log(dog.makeSound())
```
Comment (yugandhar-pathi) [inline learnClass/learnClass.js]: Deleted commented code.

File: learnClass/reviewClass.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        console.log(this.speed)
+    }
+}
+
+let children = new Children("eats slowly")
+let adults = new Adult("eats very fast")
+let old = new Old("eats fast")
+children.eatSpeedChildren()
+adults.eatSpeedAdult()
+old.eatSpeedOld()
```
Comment (yugandhar-pathi) [inline learnClass/reviewClass.js]: Remove this code and add Unit tests for this.

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: learnClass/learnClass.js (hunk 6 above)
Comment (KalyaniPayyavula) [reply learnClass/learnClass.js]: done

File: learnClass/learnClass.js (hunk 7 above)
Comment (KalyaniPayyavula) [reply learnClass/learnClass.js]: done

File: learnClass/learnClass.js (hunk 8 above)
Comment (KalyaniPayyavula) [reply learnClass/learnClass.js]: done

File: learnClass/learnClass.js (hunk 9 above)
Comment (KalyaniPayyavula) [reply learnClass/learnClass.js]: done

File: learnClass/reviewClass.js (hunk 10 above)
Comment (KalyaniPayyavula) [reply learnClass/reviewClass.js]: done

**Review by yugandhar-pathi — APPROVED** [review]
Good Job!!

**Pull request merged**